includes/class-wc-payments-payment-request-button-handler.php

```diff
@@ -531,6 +531,16 @@ public function should_show_payment_request_button() {
 			return false;
 		}
 
+		// Cart total is 0 or is on prouct page and product price is 0.
+		if (
+			( 0.0 === (float) WC()->cart->get_total( 'edit' ) && ! $this->is_product() ) ||
```

Comment:
We can make the OR checks more consistent by keeping the is product check first in both cases.

```suggestion
			( ! $this->is_product() && 0.0 === (float) WC()->cart->get_total( 'edit' ) ) ||
			( $this->is_product() && 0.0 === (float) $this->get_product()->get_price() )
```

Reply:
Also, curious what "edit" means in get_total.

Reply:
"edit" is the context. basically, if context is "view", the total will be formatted for display ( add the currency sign to the string... etc ) .

Reply:
Got it, thanks!